templates/tensorflow/r2.2/retinanet.libsonnet

```diff
@@ -0,0 +1,105 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+local base = import "base.libsonnet";
+local timeouts = import "../../timeouts.libsonnet";
+local tpus = import "../../tpus.libsonnet";
+
+{
+  local retinanet = base.GardenTest {
+    modelName: "retinanet",
+    paramsOverride:: {
+      eval: {
+        eval_file_pattern: "$(COCO_DIR)/val*",
+        batch_size: 8,
+        val_json_file: "$(COCO_DIR)/instances_val2017.json",
+      },
+      predict: {
+        predict_batch_size: 8,
+      },
+      architecture: {
+        use_bfloat16: true,
+      },
+      retinanet_parser: {
+        use_bfloat16: true,
+      },
+      train: {
+        checkpoint: {
+          path: "$(RESNET_PRETRAIN_DIR)/resnet50-checkpoint-2018-02-07",
+          prefix: "resnet50/",
+        },
+        total_steps: error "Must set `train.total_steps`",
+        batch_size: error "Must set `train.batch_size`",
+        train_file_pattern: "$(COCO_DIR)/train*",
+      },
+    },
+    command: [
+      "python3",
+      "official/vision/detection/main.py",
+      "--tpu=$(KUBE_GOOGLE_CLOUD_TPU_ENDPOINTS)",
+      "--strategy_type=tpu",
+      "--params_override=%s" % (std.manifestYamlDoc(self.paramsOverride) + "\n"),
+      "--model_dir=$(MODEL_DIR)",
+    ],
+  },
+  local convergence = base.Convergence {
+    command+: [
+      "--mode=train",
+    ],
```

Comment:
Hmm I suppose it would be non-trivial to add in the extra eval donut right?

Reply:
It should be possible, but I haven't added good support for that case in the templates